source/extensions/filters/http/aws_lambda/aws_lambda_filter.h

```diff
@@ -70,19 +72,33 @@ class Filter : public Http::PassThroughFilter, Logger::Loggable<Logger::Id::filt
   Filter(const FilterSettings& config,
          const std::shared_ptr<Extensions::Common::Aws::Signer>& sigv4_signer);
 
-  Http::FilterHeadersStatus decodeHeaders(Http::RequestHeaderMap& headers,
-                                          bool end_stream) override;
+  Http::FilterHeadersStatus decodeHeaders(Http::RequestHeaderMap&, bool end_stream) override;
   Http::FilterDataStatus decodeData(Buffer::Instance& data, bool end_stream) override;
 
-private:
+  Http::FilterHeadersStatus encodeHeaders(Http::ResponseHeaderMap&, bool end_stream) override;
+  Http::FilterDataStatus encodeData(Buffer::Instance& data, bool end_stream) override;
+
   /**
-   * Calculates the route specific Lambda ARN if any.
+   * Calculates the function ARN, value of pass-through, etc. by checking per-filter configurations
+   * and general filter configuration. Ultimately, the most specific configuration wins.
+   * @return error message if settings are invalid. Otherwise, empty string.
    */
-  absl::optional<Arn> calculateRouteArn();
+  std::string resolveSettings();
+
+private:
+  absl::optional<FilterSettings> getRouteSpecificSettings() const;
+  // Convert the HTTP request to JSON request.
+  void jsonizeRequest(const Http::RequestHeaderMap& headers, const Buffer::Instance* body,
+                      Buffer::Instance& out) const;
+  // Convert the JSON response to a standard HTTP response.
+  void dejsonizeResponse(Http::ResponseHeaderMap& headers, const Buffer::Instance& body,
+                         Buffer::Instance& out) const;
```

Comment:
Is there a reason why we are using an `Buffer::Instance pointer` in one function and an `Buffer::Instance` reference in the other?  Can we make these consistent?

Reply:
The pointer is because it's optional. You can make requests without a body and the filter should still convert that body-less request to a JSON payload and send it to Lambda.

The other one is on the response side. Lambda will always send a response body back if the HTTP status code is 2xx

Reply:
👍